Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentModeCommon.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Math/Vector3.h>
+#include <AzCore/std/string/string.h>
+#include <Editor/EditorJointCommon.h>
+
+namespace PhysX::JointsComponentModeCommon
+{
+    namespace SubComponentModes
+    {
+        enum class ModeType : AZ::u32
+        {
+            Translation = 0,
+            Rotation,
+            MaxForce,
+            MaxTorque,
+            Damping,
+            Stiffness,
+            TwistLimits,
+            SwingLimits,
+            SnapPosition,
+            SnapRotation,
+
+            NumModes
```

Comment:
suggestion: I understand this is a fairly well established technique but one big downside I see is there's now an 'invalid' state `ModeType` can technically be in (the mode being `NumModes` makes no sense). We can guard against this with asserts and what not but it is a potential source of errors. I'd be tempted to avoid adding it and instead just using a fixed size, but I understand if you disagree (this is a preference thing, just wanted to mention it so feel free to ignore this). If it stays the name `ModeCount` is potentially clearer as then no abbreviation is required.